### ISSUE TITLE
Add the ability to update MODX via widget

### ISSUE THREAD

**Issue by Ruslan-Aleev** [modxcms/revolution#16660]
### Summary
At the moment, you can't update MODX via the manager, the widget simply displays information about the new version.

And if you update manually, then downloading and replacing several thousand files is a strange solution.

We need to implement a CMS update in the manager panel, this is logical behavior of the CMS, it is strange that MODX does not have this.

### Suggested solution(s)
We can use the package code as a basis: https://extras.modx.com/package/upgrademodx, https://extras.modx.com/package/modupdater

### Related issue(s)/PR(s)
https://github.com/modxcms/revolution/issues/14591

**Comment by smg6511** [comment]
Totally agree, @Ruslan-Aleev, but let's just have one issue open on this subject (maybe close #14591 and keep this one). I had also mentioned this (namely the desire to integrate @BobRay's Upgrade MODX into the core) when I did the update to the built-in Updates widget. And, yeah, most of MODX's competitors have an integrated updater for their core app _and_ its extensions (a couple of CMSs I really dislike having to use come to mind). 

The main challenge here is that there have only been a few people actively contributing in the past couple years and there's a pretty long list of issues to tackle :-/

**Comment by Ruslan-Aleev** [comment]
There are 2 issues here, they are logically connected, but the essence is different.
I would leave both, BUT the update is a priority, yes.

**Comment by smg6511** [comment]
Oh, I see what you're saying now - in the other feature request you want to add the ability to choose which _type_ of distribution in addition to the version, correct?

**Comment by Ruslan-Aleev** [comment]
Yes, we were talking about a small php file, through which you could update MODX (by selecting a distribution), or download MODX from scratch to the server. Perhaps, if there is an update via a widget, this file will no longer be needed, or somehow integrate a file that would work both in the widget and by itself (for example, for a clean installation of MODX).

**Comment by BobRay** [comment]
Assume you know about the UpgradeMODX extra, which updates MODX from a widget. It always upgrades you to the Traditional distribution, but it works fine even if your original version was the Advanced distribution. It won't work for a clean new install, though (because it's a widget).

**Comment by Ruslan-Aleev** [comment]
Yes, an extra is also an option, but not all users install it. Also, a extra is still a separate product, and the system updates in the manager are, as it seems to me, the basis for CMS security.

**Comment by BobRay** [comment]
I don't think UpgradeMODX would be very difficult to integrate into the core. I could be wrong.
